components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/ClientAuthenticationHandler.java

```diff
@@ -143,4 +145,34 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) thr
         }
         return authenticationResult;
     }
+
+    /**
+     * Check whether default credentials are being used for ClientAuthentication.
+     *
+     * @return true if default credentials are being used.
+     */
+    public boolean isUsingDefaultCredentials() {
+
+        String hashedPasswordFromConfigFile = AuthConfigurationUtil.getInstance().getClientAuthenticationHash(defaultAppName);
+
+        if (hashedPasswordFromConfigFile != null) {
+            try {
+                MessageDigest dgst = MessageDigest.getInstance(hashingFunction);
+                byte[] byteValue = dgst.digest(defaultPassword.getBytes());
+
+                //convert the byte to hex format
+                StringBuilder sb = new StringBuilder();
+                for (byte b : byteValue) {
+                    sb.append(Integer.toString((b & 0xff) + 0x100, 16).substring(1));
+                }
+
+                if (hashedPasswordFromConfigFile.equals(sb.toString())) {
+                    return true;
+                }
+            } catch (NoSuchAlgorithmException e) {
+                throw new RuntimeException(e);
```

Comment:
Let's not break the flow on this. Let's put a warning.

Reply:
Addressed